<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [waffle_pi]"/>
  <arg name="sydney_bot"  default="sydney_bot"/>
  <arg name="alec_bot" default="alec_bot"/>
  <arg name="rachel_bot"  default="rachel_bot"/>
  <arg name="kir_bot"   default="kir_bot"/>

  <arg name="sydney_bot_x_pos" default="0.0"/>
  <arg name="sydney_bot_y_pos" default="0.0"/>
  <arg name="sydney_bot_z_pos" default="0.0"/>
  <arg name="sydney_bot_yaw"   default="0.0"/>

  <arg name="alec_bot_x_pos" default="0.0"/>
  <arg name="alec_bot_y_pos" default="-1.0"/>
  <arg name="alec_bot_z_pos" default="0.0"/>
  <arg name="alec_bot_yaw"   default="0.0"/>

  <arg name="rachel_bot_x_pos" default="0.0"/>
  <arg name="rachel_bot_y_pos" default="-2.0"/>
  <arg name="rachel_bot_z_pos" default="0.0"/>
  <arg name="rachel_bot_yaw"   default="0.0"/>

  <arg name="kir_bot_x_pos" default="1.0"/>
  <arg name="kir_bot_y_pos" default="-1.0"/>
  <arg name="kir_bot_z_pos" default="0.0"/>
  <arg name="kir_bot_yaw"   default="0.0"/>

  <arg name="color_sydney_bot" default="Gazebo/Green"/>
  <arg name="color_alec_bot" default="Gazebo/Blue"/>
  <arg name="color_rachel_bot" default="Gazebo/Yellow"/>
  <arg name="color_kir_bot" default="Gazebo/Red"/>


  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find final_project)/worlds/enclosed_area_3.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <group ns = "$(arg sydney_bot)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find final_project)/urdf/turtlebot3_$(arg model).urdf.xacro' color:=$(arg color_sydney_bot)" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg sydney_bot)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg sydney_bot) -x $(arg sydney_bot_x_pos) -y $(arg sydney_bot_y_pos) -z $(arg sydney_bot_z_pos) -Y $(arg sydney_bot_yaw) -param robot_description" />
  </group>

  <group ns = "$(arg alec_bot)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find final_project)/urdf/turtlebot3_$(arg model).urdf.xacro' color:=$(arg color_alec_bot)" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg alec_bot)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg alec_bot) -x $(arg alec_bot_x_pos) -y $(arg alec_bot_y_pos) -z $(arg alec_bot_z_pos) -Y $(arg alec_bot_yaw) -param robot_description" />
  </group>

  <group ns = "$(arg rachel_bot)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find final_project)/urdf/turtlebot3_$(arg model).urdf.xacro' color:=$(arg color_rachel_bot)" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg rachel_bot)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg rachel_bot) -x $(arg rachel_bot_x_pos) -y $(arg rachel_bot_y_pos) -z $(arg rachel_bot_z_pos) -Y $(arg rachel_bot_yaw) -param robot_description" />
  </group>

  <group ns = "$(arg kir_bot)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find final_project)/urdf/turtlebot3_$(arg model).urdf.xacro' color:=$(arg color_kir_bot)" />

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg kir_bot)" />
    </node>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model $(arg kir_bot) -x $(arg kir_bot_x_pos) -y $(arg kir_bot_y_pos) -z $(arg kir_bot_z_pos) -Y $(arg kir_bot_yaw) -param robot_description" />
  </group>

</launch>
